neo.UnitTests/Network/P2P/Payloads/UT_Transaction.cs

```diff
@@ -275,5 +275,94 @@ public void ToJson()
             jObj["script"].AsString().Should().Be("4220202020202020202020202020202020202020202020202020202020202020");
             jObj["sys_fee"].AsNumber().Should().Be(42);
         }
+
+        [TestMethod]
+        public void Infinite_Loop_Not_Allowed_On_Verification()
+        {
+            byte[] script;
+            using (ScriptBuilder sb = new ScriptBuilder())
+            {
+                sb.Emit(OpCode.JMP, new byte[] { 0, 0, 0, 0 });
+                script = sb.ToArray();
+            }
+
+            long netfee = 100000000; // network fee
+
+            // Check Application
+            long appGas = 0;
+            using (ApplicationEngine engine = new ApplicationEngine(TriggerType.Application, null, null, netfee, false))
+            {
+                engine.LoadScript(script);
+                Assert.AreEqual(VMState.FAULT, engine.Execute());
+                Assert.AreEqual(0, engine.ResultStack.Count);
+                appGas += engine.GasConsumed;
+            }
+            // many gas are spent before fault
+            Assert.AreEqual(appGas, 100000040);
```

Comment:
We are consuming more gas than limit... 40 units more here, it shouldn't I guess.

Reply:
```csharp
private bool AddGas(long gas)
        {
            GasConsumed = checked(GasConsumed + gas);
            return testMode || GasConsumed <= gas_amount;
        }
```
the method first increase the gas, so is normal to overpass this limit i think

Reply:
This is passing 40, its reasonable... the other is passing 1 million.

Reply:
We should break before adding to this variable, I guess.